components/local-app/pkg/auth/auth.go

```diff
@@ -37,21 +38,42 @@ type LoginOpts struct {
 	GitpodURL string
 }
 
+const html = `
+<html>
+	<head>
+    	<meta charset="utf-8">
+    	<title>Done</title>
+		<script>
+			if (window.opener) {
+				const message = new URLSearchParams(window.location.search).get("message");
+				window.opener.postMessage(message, "https://${window.location.hostname}");
+			} else {
+				console.log("This page was not opened by Gitpod.")
+				setTimeout("window.close();", 1000);
+			}
+		</script>
+	</head>
+	<body>
+		If this tab is not closed automatically, feel free to close it and proceed. <button type="button" onclick="window.open('', '_self', ''); window.close();">Close</button>
+	</body>
+</html>`
+
 // Login walks through the login flow for obtaining a Gitpod token
 func Login(ctx context.Context, opts LoginOpts) (token string, err error) {
-	rl, err := net.Listen("tcp", "localhost:0")
+	rl, err := net.Listen("tcp", "localhost:64110")
```

Comment:
I reckon the "use unused port" way is less error prone - or am I missing a reason to hardcode this?

Reply:
I agree, but yes, (unfortunately) the redirect URL has to be known to the backend.

Reply:
Could it be configurable if the port is already used?

Reply:
No, not yet at least. The backend validates the full redirect uri, including port, so it would need a way to override that behaviour (which might be worth PR-ing on the lib).

In the interim I've changed it, along with the backend, to try a range of 10 ports and be more explicit about using IPV4 to workaround issues with IPV6 I saw while testing.